gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -621,7 +621,8 @@ private boolean shouldStickToTheSameNode(int containerExitStatus) {
   protected void handleContainerCompletion(ContainerStatus containerStatus) {
     Map.Entry<Container, String> completedContainerEntry = this.containerMap.remove(containerStatus.getContainerId());
     if (completedContainerEntry == null) {
```

Comment:
Rather than if (completedContainerEntry == null) here, how about surrounding lines 647-676 inside if(completedContainerEntry != null) {...}? Ideally, we would still like to print out the container diagnostic log lines, and check if the container is in the releasedContainerCache.